stats/opentelemetry/internal/grpc_trace_bin_propagator.go

```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package internal
+
+import (
+	"context"
+	"encoding/base64"
+
+	otelpropagation "go.opentelemetry.io/otel/propagation"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	itracing "google.golang.org/grpc/stats/opentelemetry/internal/tracing"
+)
+
+// TODO: Move out of internal as part of open telemetry API
```

Comment:
Things should probably either be in a visible location and labeled clearly as EXPERIMENTAL, or our plans should be to keep them `internal` long-term.  I don't understand why we'd add this into `internal` and then move it out later.  That seems like unnecessary churn.

Reply:
yeah initially i thought that in case we don't have API and do a release in between, this GRPCTraceBinPropagator is an unusable exported entity visible to users. Is that ok?

So, i talked to Easwar and he suggested we can have it internally until then and move out along with Otel API change PR.